client/signup/jetpack-connect/index.jsx

```diff
@@ -136,8 +131,17 @@ class JetpackConnectMain extends Component {
 			this.props.jetpackConnectSite.isFetching;
 	}
 
-	getCurrentUrl() {
-		let url = this.refs.siteUrlInputRef.state.value.toLowerCase();
+	handleUrlChange = ( event ) => {
+		const url = event.target.value;
+		this.setState( {
+			currentUrl: this.cleanUrl( url ),
+			shownUrl: url,
+		} );
+		this.dismissUrl();
```

Comment:
`dismissUrl` depends on `state.currentUrl` internally, so this call should either be in the callback to `setState` or before `setState`. I need to have a look to see if the previous or new URL should be the argument to `this.props.dismissUrl`.